lib/tessel/update.js

```diff
@@ -53,33 +64,36 @@ Tessel.prototype.updateOpenWRT = function(image) {
       });
     })
     .then(() => {
-      return new Promise((resolve) => {
+      return new Promise((resolve, reject) => {
         // Begin the sysupgrade
         logs.info('Starting OpenWRT update. Please do not remove power from Tessel.');
         // The USBDaemon will cut out or the SSH command will close
-        this.connection.exec(commands.sysupgrade(updatePath), (err, remoteProc) => {
-          remoteProc.stdout.on('data', function(d) {
-            if (d.toString().includes('Upgrade completed')) {
-              resolve();
-            }
-          });
+        this.connection.exec(commands.sysupgrade(updatePath), (err) => {
+          if (err) {
+            return reject(err);
+          } else {
+            return resolve();
+          }
         });
       });
     });
 };
 
 Tessel.prototype.updateFirmware = function(image) {
-  logs.info('Updating firwmare (2/2)');
-  // This must be USB connection
+  return new Promise((resolve, reject) => {
+    logs.info('Updating firmware (2/2)');
```

Comment:
I was just having issues with Promises resolving when I thought they should. I can revert this change if need be.